### ISSUE TITLE
feat: auto register tracing layer

### ISSUE THREAD

**Issue by chesedo** [shuttle-hq/shuttle#280]
Currently `log` works transparently for [rocket with shuttle](https://github.com/shuttle-hq/shuttle/blob/4785a54ae4cda967a20425206087d76d11605b90/codegen/src/lib.rs#L158-L160), but other frameworks might need to register a middleware for logs.

We are moving over to tracing in #267. Once the move is done, we should  automatically register the tracing layer for frameworks that need it by [passing the tracing struct to bind](https://github.com/shuttle-hq/shuttle/blob/4785a54ae4cda967a20425206087d76d11605b90/service/src/lib.rs#L341) and registering it correctly for each framework

**Comment by nahuakang** [comment]
This sounds interesting (I've never done this aspect before). Would it be okay if I claim it (happy to also let it pass to another contributor whom you deem fit, too) :)

**Comment by chesedo** [comment]
@nahuakang I've been able to confirm this will be possible if you still want to take this?

**Comment by chesedo** [comment]
This [tmp commit](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071) will be a good starting point for what is needed.

The [poem hello_world example](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071#diff-fac52dbacd03c774191a9602d3a0d99c5a7444304506e7486f1ed7e6ce6eef1bR21) is used to prototype the changes that will be needed for `codegen`. These charges appear to be [changing](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071#diff-fac52dbacd03c774191a9602d3a0d99c5a7444304506e7486f1ed7e6ce6eef1bR25) the `Box<dyn log::Log>` to a `tracing_subsriber::Layer` for registration with the tracing registry. It also [confirms the implementation work for both](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071#diff-fac52dbacd03c774191a9602d3a0d99c5a7444304506e7486f1ed7e6ce6eef1bR6-R8) `log` and `tracing`.

This [`LogItem`](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071#diff-d38884f7391ae50f795707dbb281656e8afe38ea3ecac15a6ee80d52724017c0R210) should probably be replaced with the new [`Item`](https://github.com/shuttle-hq/shuttle/pull/298/files#diff-ee89ad23b6dbfadf93921511dc39cb7a8d8744d1099809c65eee61b86b9d156cR22) coming from `deployer`. Then the [logger will need to be updated](https://github.com/shuttle-hq/shuttle/commit/72ad972e32c5fafbdf7b5abc1b6f8b91abfc5071) to send this new `Item`. These [lines from the deploy_layer](https://github.com/shuttle-hq/shuttle/blob/c0119682f97e3aba68457e2a1b43ad1219ba0c18/deployer/src/deployment/deploy_layer.rs#L190-L204) might give you some inspiration for constructing an `Item`. It should be safe to ignore the `tracing` span/scope for now as we can take care of it in a separate PR.

I was able to test this tmp commit with the following command in the `examples/poem/hello_world` directory:
```bash
cargo run --manifest-path ../../../Cargo.toml --bin cargo-shuttle -- run
```

**Comment by nahuakang** [comment]
Still happy to take this if there's no urgent deadline as I can start on this maybe in 7-14 days :)

**Comment by chesedo** [comment]
Sounds good to me... I guess that means anyone else has about 7 days to claim this if they want it :grin:

**Comment by coszio** [comment]
If it is not too late I'd like to take this, but the tracing migration PR indicates it is still open. Is this still waiting for it?

**Comment by chesedo** [comment]
@coszio feel free to start on this one.

This one is a bit different from the open PR: this one is about changing the registration from `log` to `tracing` inside the codegen for services deployed with shuttle; the open PR is about changing shuttle itself (ie the code that deploys shuttle services) from `log` to `tracing`
